### PR TITLE
Add `Node` driver suite tests

### LINKED ISSUE #226: Improve test coverage
# Summary

Add tests for `node.go` and improve the test coverage.

### PR DESCRIPTION
**Pull request by sujeet01**
# Proposed changes

- Add `node` driver suite tests  
- Improve `controller` tests
- Add make target for mock generate using `mockgen`
- Add `update-gomock` script to update mountutil mock to fix the non-exported method in `mount.Interface`
- Refactor `driver` struct to use wrapper interfaces of `os` and `mount` utils
- Mock `mount` and `os` operations

Fixes #226

### CODE REVIEW

File: hack/update-gomock [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Fixes "MountWrapper Type cannot implement 'MountWrapper' as it has a non-exported method and is defined in a different package"
+# See https://github.com/kubernetes/mount-utils/commit/a20fcfb15a701977d086330b47b7efad51eb608e for context.
+sed -i '/type MockMountWrapper struct {/a \\mount.Interface' pkg/driver/mocks/mock_mountutil.go
```
Comment (afritzler) [inline hack/update-gomock:7]: Just use `mountutil.go` as the filename.

**Review by afritzler — CHANGES_REQUESTED** [review]
Thanks a lot for the PR @sujeet01! Please find my review above. How about calling `mountutil` `mountutils` and `osutil` `osutils` as they embody a collection of utility functions.

File: Makefile [hunk 2]
```diff
@@ -158,3 +160,16 @@ $(ADDLICENSE): $(LOCALBIN)
...
+generate: mockgen ## Generate code (mocks etc.).
+	MOCKGEN=$(MOCKGEN) go generate ./...
+
+.PHONY: mockgen
+mockgen: $(MOCKGEN)
+$(MOCKGEN): $(LOCALBIN)
+	test -s $(LOCALBIN)/mockgen || GOBIN=$(LOCALBIN) go install github.com/golang/mock/mockgen@$(MOCKGEN_VERSION)
+
+.PHONY: update-mock
+update-mock:
```
[the remark this answers is not in the document]
Comment (sujeet01) [reply Makefile]: The `update-mock` target is used to update the generated mock with the script `/hack/update-gomock`. 

I have renamed the `generate` target to `generate-mocks`

File: pkg/driver/driver.go [hunk 3]
```diff
@@ -16,19 +16,22 @@ package driver
...
 	"k8s.io/klog/v2"
-	"k8s.io/mount-utils"
-	utilexec "k8s.io/utils/exec"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type driver struct {
-	mountUtil     *mount.SafeFormatAndMount
+	mountUtil     mountutils.MountWrapper
+	osUtil        osutils.OSWrapper
```
Comment (afritzler) [inline pkg/driver/driver.go:34]: Just call those two fields `mount` and `os`.

File: Makefile [hunk 4]
```diff
@@ -158,3 +160,16 @@ $(ADDLICENSE): $(LOCALBIN)
...
+	MOCKGEN=$(MOCKGEN) go generate ./...
+
+.PHONY: mockgen
+mockgen: $(MOCKGEN)
+$(MOCKGEN): $(LOCALBIN)
+	test -s $(LOCALBIN)/mockgen || GOBIN=$(LOCALBIN) go install github.com/golang/mock/mockgen@$(MOCKGEN_VERSION)
+
+.PHONY: update-mock
+update-mock: ## Run update-gomock script to update mountutils mock file
+	./hack/update-gomock
```
Comment (afritzler) [inline Makefile:175]: Wouldn't it make sense to move this before we run `go generate ...` in the `generate-mocks` directive?

**Review by afritzler — CHANGES_REQUESTED** [review]

File: pkg/driver/driver.go (hunk 3 above)
Comment (sujeet01) [reply pkg/driver/driver.go]: Done

File: Makefile (hunk 4 above)
Comment (sujeet01) [reply Makefile:175]: `update-mock` modifies the mock generated by `generate-mocks` so it should be called after `generate-mocks`

**Pull request merged**